pkg/utils/database/mysql.go

```diff
@@ -68,7 +68,8 @@ func (m Mysql) getDbConn(user, password string) (*sql.DB, error) {
 
 	switch m.Backend {
 	case "google":
-		db, err = mysql.DialPassword(m.Host, user, password)
+		// TODO: DialPassword is deprecated, it should be gone
```

Comment:
https://github.com/GoogleCloudPlatform/cloud-sql-go-connector/tree/v1.4.5#mysql

When google is used, we have already 2 credentials files mounted to the db-operator. So we could just us them and take a basic mysql lib. 

alternative we would then also be able to extend this to use Workload Identity, then we could go away from mounting files at all and make the setup more streamline.

another alternative would be, to also have instance `google` and `google-internal`. for the internal, we would not need a cloud proxy at all, but need to connect to the private IP and not to the instance name (public). And for maybe Security reasons (think the cloudproxy public connection should be secure/fine, but from 2 audits ... sec companies don't like this), deprecate the public connection `google` then?

Reply:
Attached this comment to the issue: https://github.com/db-operator/db-operator/issues/73